infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServer.java

```diff
@@ -552,6 +562,29 @@ private void handleTxToken(CorfuPayloadMsg<TokenRequest> msg, ChannelHandlerCont
         handleAllocation(msg, ctx, r);
     }
 
+    /**
+     * this method serves a monotonically increasing counter as an Ordered GUID.
+     * It simply returns a counter along with the current epoch value.
+     * If the sequencer restarts the epoch ticks up while the counter resets, thereby
+     * allowing a globally unique monotonically increasing sequence to be served.
+     *
+     * @param msg corfu message containing allocation
+     * @param ctx netty ChannelHandlerContext
+     * @param r   server router
+     */
+    private void handleOrderedGuidToken(CorfuPayloadMsg<TokenRequest> msg,
+                                        ChannelHandlerContext ctx, IServerRouter r) {
+        final TokenRequest req = msg.getPayload();
+
+        Token token;
+        synchronized (globalSequence) {
```

Comment:
Sequencer is a single threaded, so you don't need `synchronized`

Reply:
that explains a lot. thank you!

Reply:
thanks! I will modify it as you suggest.